javaslang-benchmark/src/test/java/javaslang/benchmark/ArrayBenchmark.java

```diff
@@ -0,0 +1,129 @@
+    public static class Base {
+        @Param({ "10", "100", "1000" })
+        public int CONTAINER_SIZE;
+
+        public Integer[] ELEMENTS;
+
+        @Setup
+        public void setup() {
+            final Random random = new Random(0);
+
+            ELEMENTS = new Integer[CONTAINER_SIZE];
+            for (int i = 0; i < CONTAINER_SIZE; i++) {
+                final int value = random.nextInt(CONTAINER_SIZE) - (CONTAINER_SIZE / 2);
+                ELEMENTS[i] = value;
+            }
+        }
+
+        protected static <T> void assertEquals(T a, T b) {
+            if (!Objects.equals(a, b)) {
+                throw new IllegalStateException(a + " != " + b);
+            }
+        }
+    }
+
+    public static class AddAll extends Base {
+        @Benchmark
+        @SuppressWarnings("ManualArrayCopy")
+        public void java_mutable() {
+            final Integer[] values = new Integer[ELEMENTS.length];
+            for (int i = 0; i < ELEMENTS.length; i++) {
+                values[i] = ELEMENTS[i];
+            }
+            assertEquals(values.length, CONTAINER_SIZE);
+        }
+
+        @Benchmark
+        public void slang_persistent() {
+            javaslang.collection.Array<Integer> values = javaslang.collection.Array.empty();
+            for (Integer element : ELEMENTS) {
+                values = values.append(element);
+            }
+            assertEquals(values.size(), CONTAINER_SIZE);
+        }
+    }
+
+    public static class Iterate extends Base {
+        @State(Scope.Thread)
+        public static class Initialized {
+            Integer[] javaMutable;
+
+            int expectedAggregate = 0;
+            javaslang.collection.Array<Integer> slangPersistent = javaslang.collection.Array.empty();
+
+            @Setup(Level.Invocation)
+            public void initializeMutable(Base state) {
+                assertEquals(javaMutable, null);
+                javaMutable = state.ELEMENTS.clone();
+                assertEquals(javaMutable.length, state.CONTAINER_SIZE);
+
+                if (expectedAggregate == 0) {
+                    for (Integer element : state.ELEMENTS) {
+                        expectedAggregate ^= element;
+                    }
+
+                    assertEquals(slangPersistent.size(), 0);
+                    for (Integer element : state.ELEMENTS) {
+                        slangPersistent = slangPersistent.prepend(element);
+                    }
+                    assertEquals(slangPersistent.size(), state.CONTAINER_SIZE);
+                }
+            }
+
+            @TearDown(Level.Invocation)
+            public void tearDown() {
+                javaMutable = null;
+            }
+        }
+
+        @Benchmark
+        @SuppressWarnings("ForLoopReplaceableByForEach")
+        public void java_mutable(Initialized state) {
+            int aggregate = 0;
+            for (int i = 0; i < CONTAINER_SIZE; i++) {
+                aggregate ^= state.javaMutable[i];
```

Comment:
xor is faster than adding to a list, so it doesn't distort the ratios and provides a very basic validation also